frontend/src/components/organisms/FormRouter.js

```diff
@@ -1,24 +1,36 @@
+import { isEmpty } from 'lodash';
 import React from 'react';
 import { useParams } from 'react-router-dom';
-import { DATA_PROVIDER_PARAMETERS } from '../../config/data-provider-parameters';
-import { AuthRequired } from './AuthContext';
+import Loader from '../atoms/Loader';
+import Enrollment from '../templates/Enrollment';
+import { useFullDataProvider } from '../templates/hooks/use-full-data-provider';
 import NotFound from './NotFound';
 
 const FormRouter = () => {
   const { targetApi } = useParams();
 
-  const TargetApiComponent =
-    DATA_PROVIDER_PARAMETERS[targetApi.replace(/-/g, '_')]?.component;
+  const { Component, configuration, notFound } = useFullDataProvider({
+    targetApi: targetApi.replace(/-/g, '_'),
```

Comment:
This method `targetApi.replace(/-/g, '_')` is used three times in the same document, maybe it can be useful to make it a function